metaserver/src/main/scala/scala/meta/languageserver/search/InMemorySymbolIndex.scala

```diff
@@ -162,39 +163,40 @@ class InMemorySymbolIndex(
    *                 - filename must be formatted as a URI
    *                 - names must be sorted
    */
-  def indexDocument(document: s.Document): Effects.IndexSemanticdb = {
-    val uri = Uri(document.filename)
-    val input = Input.VirtualFile(document.filename, document.contents)
+  def indexDocument(document: s.TextDocument): Effects.IndexSemanticdb = {
+    val uri = Uri(document.uri)
+    val input = Input.VirtualFile(document.uri, document.text)
     documentIndex.putDocument(uri, document)
-    document.names.foreach {
+    document.occurrences.foreach {
       // TODO(olafur) handle local symbols on the fly from a `Document` in go-to-definition
       // local symbols don't need to be indexed globally, by skipping them we should
       // def isLocalSymbol(sym: String): Boolean =
       // !sym.endsWith(".") &&
       //     !sym.endsWith("#") &&
       //     !sym.endsWith(")")
       // be able to minimize the size of the global index significantly.
-      //      case s.ResolvedName(_, sym, _) if isLocalSymbol(sym) => // Do nothing, local symbol.
-      case s.ResolvedName(Some(s.Position(start, end)), sym, true) =>
+      //      case s.SymbolOccurrence(_, sym, _) if isLocalSymbol(sym) => // Do nothing, local symbol.
```

Comment:
The changes for local symbols will require us to implement this TODO. Local symbols must now either use reference equality or include the original document in their syntax because two local symbols from different source files can have identical structures.